src/Microsoft.AspNet.Mvc.Core/ApplicationModels/DefaultControllerModelBuilder.cs

```diff
@@ -126,6 +130,12 @@ protected virtual ControllerModel CreateControllerModel([NotNull] TypeInfo typeI
             AddRange(controllerModel.Filters, attributes.OfType<IFilter>());
             AddRange(controllerModel.RouteConstraints, attributes.OfType<IRouteConstraintProvider>());
 
+            var policy = AuthorizationPolicy.Combine(_authorizationOptions, attributes.OfType<AuthorizeAttribute>());
+            if (policy != null)
+            {
+                controllerModel.Filters.Add(new AuthorizeFilter(policy));
```

Comment:
Instead of creating the AuthorizeFilter here, probably we should create a filter factory. That would automatically create the filter at a later point.  Also creating the factory here will enable you to not take here in ```DefaultControllerModelBuilder``` and ```DefaultActionModelBuilder```. Please see the comment on AuthorizeFilter as well.